document-store/src/main/java/org/hypertrace/core/documentstore/expression/LiteralExpression.java

```diff
@@ -0,0 +1,13 @@
+package org.hypertrace.core.documentstore.expression;
+
+import lombok.Value;
+
+/**
+ * Expression representing either a literal (or a column name)
+ *
+ * <p>Example: LiteralExpression.of("col1");
+ */
+@Value(staticConstructor = "of")
+public class LiteralExpression implements Groupable, Projectable, Sortable {
```

Comment:
Can bring over the comment about IdentifyingExpression or ColumnExpression here.